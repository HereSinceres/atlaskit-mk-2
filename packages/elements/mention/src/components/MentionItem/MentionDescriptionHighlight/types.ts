import { MentionDescription } from '../../../types';

export interface Props {
  mention: MentionDescription;
}
