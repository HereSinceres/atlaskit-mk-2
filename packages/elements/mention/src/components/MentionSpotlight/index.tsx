import * as React from 'react';
import * as Styled from './styles';

import Button from '@atlaskit/button';
import EditorCloseIcon from '@atlaskit/icon/glyph/editor/close';

import Tooltip from '@atlaskit/tooltip';
import {
  SpotlightTitle,
  SpotlightCloseTooltip,
  SpotlightDescription,
  SpotlightDescriptionLink,
} from '../../util/i18n';

export interface Props {
  createTeamLink: string;
  /** Callback to track the event where user click on x icon */
  onClose: () => void;
}

export default class MentionSpotlight extends React.Component<Props, {}> {
  render() {
    const { onClose, createTeamLink } = this.props;

    return (
      <Styled.Card>
        <Styled.Content>
          <Styled.Aside>
            <img src={Styled.iconUrl} height={32} />
          </Styled.Aside>
          <Styled.Section>
            <Styled.Heading>
              <Styled.Title>
                <SpotlightTitle />
              </Styled.Title>
            </Styled.Heading>
            <Styled.Body>
              <SpotlightDescription>
                {description => (
                  <p>
                    {description}
                    <SpotlightDescriptionLink>
                      {linkText => (
                        <a href={createTeamLink} target="_blank">
                          {' '}
                          {linkText}{' '}
                        </a>
                      )}
                    </SpotlightDescriptionLink>
                  </p>
                )}
              </SpotlightDescription>
            </Styled.Body>
          </Styled.Section>
          <Styled.Actions>
            <SpotlightCloseTooltip>
              {tooltip => (
                <Tooltip content={tooltip} position="bottom">
                  <Button
                    appearance="subtle"
                    iconBefore={<EditorCloseIcon label="Close" size="medium" />}
                    onClick={onClose}
                    spacing="none"
                  />
                </Tooltip>
              )}
            </SpotlightCloseTooltip>
          </Styled.Actions>
        </Styled.Content>
      </Styled.Card>
    );
  }
}
