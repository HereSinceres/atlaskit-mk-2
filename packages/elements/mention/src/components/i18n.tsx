import * as React from 'react';
import {
  defineMessages,
  IntlProvider,
  injectIntl,
  InjectedIntlProps,
} from 'react-intl';

import * as untypedI18n from '../i18n';

export const messages = defineMessages({
  noAccessWarning: {
    id: 'fabric.mention.noAccess.warning',
    defaultMessage: "{name} won't be notified as they have no access",
    description:
      "Warning message to show that the mentioned user won't be notified",
  },
  noAccessLabel: {
    id: 'fabric.mention.noAccess.label',
    defaultMessage: 'No access',
    description: 'Label for no access icon',
  },
  defaultHeadline: {
    id: 'fabric.mention.error.defaultHeadline',
    defaultMessage: 'Something went wrong',
    description:
      'Error message shown when there is an error communicating with backend',
  },
  defaultAdvisedAction: {
    id: 'fabric.mention.error.defaultAction',
    defaultMessage: 'Try again in a few seconds',
    description: 'Default advised action when an error occurs',
  },
  loginAgain: {
    id: 'fabric.mention.error.loginAgain',
    defaultMessage: 'Try logging out then in again',
    description:
      'Login again message when there is an authentication error occurs',
  },
  differentText: {
    id: 'fabric.mention.error.differentText',
    defaultMessage: 'Try entering different text',
    description: 'Enter different text message when a forbidden error occurs',
  },
  memberCountWithoutYou: {
    id: 'fabric.elements.mentions.team.member.count',
    defaultMessage: '{0, plural, one {1 member} other {{0} members}}',
    description:
      'Byline to show the number of members in the team when the current user is not a member of the team',
  },
  memberCountWithYou: {
    id: 'fabric.elements.mentions.team.member.count.including.you',
    defaultMessage:
      '{0, plural, one { 1 member} other {{0} members}}, including you',
    description:
      'Byline to show the number of members in the team when the current user is also a member of the team',
  },
  plus50MembersWithoutYou: {
    id: 'fabric.elements.mentions.team.member.50plus',
    defaultMessage: '50+ members',
    description:
      'Byline to show the number of members in the team when the number exceeds 50',
  },
  plus50MembersWithYou: {
    id: 'fabric.elements.mentions.team.member.50plus.including.you',
    defaultMessage: '50+ members, including you',
    description:
      'Byline to show the number of members in the team when the number exceeds 50 and also includes the current user',
  },
});

const i18n: { [index: string]: Object | undefined } = untypedI18n;

const getCodesFromLocale = (locale: string) => {
  const [, language, country] = /([a-z]*)[_-]?([A-Z]*)/i.exec(locale || '');
  return [language.toLowerCase(), country.toUpperCase()];
};

interface ComponentProps {
  children: React.ReactElement<any>;
}

export const MentionIntlProvider = injectIntl<ComponentProps>(
  (props: ComponentProps & InjectedIntlProps) => {
    const { children, intl } = props;
    const [language] = getCodesFromLocale(intl.locale.toString());
    const messagesByLocale = i18n[language] || i18n.en;

    return (
      <IntlProvider messages={messagesByLocale} locale={language}>
        {children}
      </IntlProvider>
    );
  },
);
