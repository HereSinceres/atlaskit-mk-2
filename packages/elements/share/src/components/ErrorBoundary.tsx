import * as React from 'react';
import { injectIntl, InjectedIntlProps } from 'react-intl';
import {
  AnalyticsEventPayload,
  WithAnalyticsEventProps,
  withAnalyticsEvents,
} from '@atlaskit/analytics-next';
import { Reason } from '../types';
import { errorToReason } from './utils';

type ErrorBoundaryProps = InjectedIntlProps & WithAnalyticsEventProps;

type ErrorBoundaryState = {
  hasError: boolean;
  reason?: Reason;
};

class ErrorBoundary extends React.Component<
  ErrorBoundaryProps,
  ErrorBoundaryState
> {
  state: ErrorBoundaryState = { hasError: false };

  fireOperationalEvent = (payload: AnalyticsEventPayload) => {
    if (this.props.createAnalyticsEvent) {
      this.props.createAnalyticsEvent(payload).fire('fabric-elements');
    }
  };

  componentDidCatch(error: any) {
    const reason = errorToReason(error);

    this.setState({
      hasError: true,
      reason,
    });
  }

  render() {
    const { hasError } = this.state;
    if (hasError) {
      // Silently fail.
      return null;
    }
    return this.props.children;
  }
}

export default withAnalyticsEvents()(injectIntl(ErrorBoundary));
