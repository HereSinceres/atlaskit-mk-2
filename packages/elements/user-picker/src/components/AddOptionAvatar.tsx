import { Skeleton } from '@atlaskit/icon';
import InviteTeamIcon from '@atlaskit/icon/glyph/invite-team';
import * as React from 'react';
import styled from 'styled-components';

const AddOptionAvatarWrapper = styled.span`
  color: black;
  padding: 2px;
`;

// This is a workaround for using Skeleton
// as Skeleton is preset with opacity, and altering the fill-color in hover state
const WhiteBackgroundWrapper = styled.div`
  background-color: white;
  border-radius: 50%;
`;

export type AddOptionAvatarProps = {
  label: string;
  size?: 'small' | 'large';
};

export const AddOptionAvatar: React.StatelessComponent<
  AddOptionAvatarProps
> = ({ size, label }) => (
  <AddOptionAvatarWrapper>
    <WhiteBackgroundWrapper>
      <Skeleton size={size}>
        <InviteTeamIcon label={label} size={size} primaryColor="white" />
      </Skeleton>
    </WhiteBackgroundWrapper>
  </AddOptionAvatarWrapper>
);

AddOptionAvatar.defaultProps = {
  size: 'large',
};
