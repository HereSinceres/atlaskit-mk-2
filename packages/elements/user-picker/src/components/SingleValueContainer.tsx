import * as React from 'react';
import { gridSize, math } from '@atlaskit/theme';
import styled from 'styled-components';
import { components, ValueContainerProps } from '@atlaskit/select';
import { SizeableAvatar } from './SizeableAvatar';
import { BORDER_PADDING } from './styles';
import { User, Option } from '../types';

const PlaceholderIconContainer = styled.div`
  padding-left: ${BORDER_PADDING}px;
  line-height: 0;
`;

const AvatarWithChildrenWrapper = styled.div`
  display: flex;
  align-items: center;
  align-content: center;
  flex: 1 1 auto;
  height: ${math.multiply(gridSize, 5)}px;
`;

const showUserAvatar = (inputValue?: string, value?: Option<User>) =>
  value && value.data && inputValue === value.label;

export class SingleValueContainer extends React.Component<
  ValueContainerProps<Option<User>>
> {
  private renderAvatar = () => {
    const {
      hasValue,
      selectProps: { appearance, isFocused, inputValue, value },
    } = this.props;

    if (isFocused || !hasValue) {
      return (
        <SizeableAvatar
          appearance={appearance}
          src={
            showUserAvatar(inputValue, value as Option<User>)
              ? (value as Option<User>).data.avatarUrl
              : undefined
          }
        />
      );
    }
    return null;
  };

  render() {
    const { children, ...valueContainerProps } = this.props;

    return (
      <components.ValueContainer {...valueContainerProps}>
        <AvatarWithChildrenWrapper>
          <PlaceholderIconContainer>
            {this.renderAvatar()}
          </PlaceholderIconContainer>
          {children}
        </AvatarWithChildrenWrapper>
      </components.ValueContainer>
    );
  }
}
