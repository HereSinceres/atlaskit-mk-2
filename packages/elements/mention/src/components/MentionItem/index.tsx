import Avatar from '@atlaskit/avatar';
import LockCircleIcon from '@atlaskit/icon/glyph/lock-circle';
import Lozenge from '@atlaskit/lozenge';
import { colors } from '@atlaskit/theme';
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import {
  HighlightDetail,
  isRestricted,
  MentionDescription,
  OnMentionEvent,
  Presence,
  UserType,
} from '../../types';
import { NoAccessLabel } from '../../util/i18n';
import { leftClick } from '../../util/mouse';
import { NoAccessTooltip } from '../NoAccessTooltip';
import {
  AccessSectionStyle,
  AvatarStyle,
  FullNameStyle,
  InfoSectionStyle,
  MentionItemStyle,
  NameSectionStyle,
  NicknameStyle,
  RowStyle,
  TimeStyle,
  TeamInformationStyle,
} from './styles';

type ReactComponentConstructor = new (props: any) => React.Component<any, any>;

interface Part {
  value: string;
  matches: boolean;
}

function renderHighlight(
  ReactComponent: ReactComponentConstructor,
  value?: string,
  highlights?: HighlightDetail[],
  prefix?: string,
) {
  if (!value) {
    return null;
  }

  const parts: Part[] = [];
  const prefixText = prefix || '';
  let lastIndex = 0;

  if (highlights) {
    for (let i = 0; i < highlights.length; i++) {
      const h = highlights[i];
      const start = h.start;
      const end = h.end;
      if (start > lastIndex) {
        parts.push({
          value: value.substring(lastIndex, start),
          matches: false,
        });
      }
      parts.push({
        value: value.substring(start, end + 1),
        matches: true,
      });
      lastIndex = end + 1;
    }
    if (lastIndex < value.length) {
      parts.push({
        value: value.substring(lastIndex, value.length),
        matches: false,
      });
    }
  } else {
    parts.push({
      value,
      matches: false,
    });
  }

  return (
    <ReactComponent>
      {prefixText}
      {parts.map((part, index) => {
        if (part.matches) {
          return <b key={index}>{part.value}</b>;
        }
        return part.value;
      })}
    </ReactComponent>
  );
}

function renderTeamInformation(
  ReactComponent: ReactComponentConstructor,
  memberCount?: number,
  includesYou?: boolean,
) {
  // todo - refactor with TeamOption ?
  // if Member count is missing, do not show the byline, regardless of the availability of includesYou
  if (memberCount === null || typeof memberCount === 'undefined') {
    return undefined;
  }
  return (
    <ReactComponent>
      <FormattedMessage
        {...(memberCount > 50 ? messages.plus50Members : messages.memberCount)}
        values={{ count: memberCount, includes: includesYou }}
      />
    </ReactComponent>
  );
}

function renderLozenge(lozenge?: string) {
  if (lozenge) {
    return <Lozenge>{lozenge}</Lozenge>;
  }
  return null;
}

function renderTime(time?: string) {
  if (time) {
    return <TimeStyle>{time}</TimeStyle>;
  }
  return null;
}

export interface Props {
  mention: MentionDescription;
  selected?: boolean;
  onMouseMove?: OnMentionEvent;
  onSelection?: OnMentionEvent;
}

export default class MentionItem extends React.PureComponent<Props, {}> {
  // internal, used for callbacks
  private onMentionSelected = (event: React.MouseEvent<any>) => {
    if (leftClick(event) && this.props.onSelection) {
      event.preventDefault();
      this.props.onSelection(this.props.mention, event);
    }
  };

  private onMentionMenuItemMouseMove = (event: React.MouseEvent<any>) => {
    if (this.props.onMouseMove) {
      this.props.onMouseMove(this.props.mention, event);
    }
  };

  render() {
    const { mention, selected } = this.props;
    const {
      id,
      highlight,
      avatarUrl,
      presence,
      name,
      mentionName,
      nickname,
      lozenge,
      accessLevel,
      context,
      userType,
    } = mention;
    const { status, time } = presence || ({} as Presence);
    const restricted = isRestricted(accessLevel);

    const nameHighlights = highlight && highlight.name;
    const nicknameHighlights = highlight && highlight.nickname;
    const borderColor = selected ? colors.N30 : undefined;

    let isTeamType: boolean;

    if (userType) {
      isTeamType = userType === UserType[UserType.TEAM];
    } else {
      isTeamType = false;
    }

    let bottomHighlight;
    if (isTeamType) {
      const includesYou = context && context.includesYou;
      const memberCount = context && context.memberCount;
      bottomHighlight = renderTeamInformation(
        TeamInformationStyle,
        memberCount,
        includesYou,
      );
    } else {
      bottomHighlight = renderHighlight(
        NicknameStyle,
        nickname,
        nicknameHighlights,
        '@',
      );
    }

    return (
      <MentionItemStyle
        selected={selected}
        onMouseDown={this.onMentionSelected}
        onMouseMove={this.onMentionMenuItemMouseMove}
        data-mention-id={id}
        data-mention-name={mentionName}
      >
        <RowStyle>
          <AvatarStyle restricted={restricted}>
            <Avatar
              src={avatarUrl}
              size="medium"
              presence={status}
              borderColor={borderColor}
            />
          </AvatarStyle>
          <NameSectionStyle restricted={restricted}>
            {renderHighlight(FullNameStyle, name, nameHighlights)}
            {bottomHighlight}
          </NameSectionStyle>
          <InfoSectionStyle restricted={restricted}>
            {renderLozenge(lozenge)}
            {renderTime(time)}
          </InfoSectionStyle>
          {restricted ? (
            <NoAccessTooltip name={name!}>
              <AccessSectionStyle>
                <NoAccessLabel>
                  {text => (
                    <LockCircleIcon label={text as string} />
                  ) /* safe to cast to string given there is no value binding */}
                </NoAccessLabel>
              </AccessSectionStyle>
            </NoAccessTooltip>
          ) : null}
        </RowStyle>
      </MentionItemStyle>
    );
  }
}

const messages = {
  memberCount: {
    id: 'fabric.elements.user-picker.team.member.count',
    defaultMessage:
      '{count} {count, plural, one {member} other {members}}{includes, select, true {, including you} other {}}',
    description:
      'Number of members in the team and whether it includes the current user',
  },
  plus50Members: {
    id: 'fabric.elements.user-picker.team.member.50plus',
    defaultMessage:
      '50+ members{includes, select, true {, including you} other {}}',
    description:
      'Number of members in a team exceeds 50 and whether it includes the current user',
  },
};
