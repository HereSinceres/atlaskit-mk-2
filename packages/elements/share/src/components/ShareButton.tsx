import Button, { ButtonAppearances } from '@atlaskit/button';
import ShareIcon from '@atlaskit/icon/glyph/share';
import * as React from 'react';
import { injectIntl, InjectedIntlProps } from 'react-intl';
import { messages } from '../i18n';

export type Props = {
  appearance?: ButtonAppearances;
  isLoading?: boolean;
  isSelected?: boolean;
  isDisabled?: boolean;
  isLoading?: boolean;
  onClick: (e: React.MouseEvent<HTMLElement>) => void;
  text?: React.ReactNode;
};

export const ShareButton: React.StatelessComponent<
  Props & InjectedIntlProps
> = ({ intl: { formatMessage }, text, ...props }) => (
  <Button
    {...props}
    iconBefore={
      <ShareIcon label={formatMessage(messages.shareTriggerButtonIconLabel)} />
    }
  >
    {text}
  </Button>
);

export default injectIntl(ShareButton);
