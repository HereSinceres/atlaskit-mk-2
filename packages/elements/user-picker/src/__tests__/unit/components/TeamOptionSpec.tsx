import { shallow } from 'enzyme';
import * as React from 'react';
import { TeamOption, TeamOptionProps } from '../../../components/TeamOption';
import { Team } from '../../../types';

describe('Team Option', () => {
  const basicTeam: Team = {
    id: 'team-7',
    name: 'Team-1',
    avatarUrl: 'https://avatars.atlassian.com/team-1.png',
    type: 'team',
  };

  const shallowOption = (props: Partial<TeamOptionProps> = {}, team: Team) =>
    shallow(<TeamOption team={team} isSelected={false} {...props} />);

  const buildTeam = (teamData: Partial<Team> = {}): Team => {
    return {
      ...basicTeam,
      ...teamData,
    };
  };

  it('should not render byline if member count is undefined', () => {
    const component = shallowOption(
      { isSelected: true },
      buildTeam({ includesYou: true }),
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correct byline if includesYou is undeifned and memberCount <= 50', () => {
    const component = shallowOption(
      { isSelected: true },
      buildTeam({
        memberCount: 45,
      }),
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correct byline if includesYou is undeifned and memberCount > 50', () => {
    const component = shallowOption(
      { isSelected: true },
      buildTeam({
        memberCount: 51,
      }),
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correct byline if includesYou = false and memberCount <= 50', () => {
    const component = shallowOption(
      { isSelected: true },
      buildTeam({
        includesYou: false,
        memberCount: 45,
      }),
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correct byline if includesYou = false and memberCount > 50', () => {
    const component = shallowOption(
      { isSelected: true },
      buildTeam({
        includesYou: false,
        memberCount: 51,
      }),
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correct byline if includesYou = true and memberCount <= 50', () => {
    const component = shallowOption(
      { isSelected: true },
      buildTeam({
        includesYou: true,
        memberCount: 45,
      }),
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correct byline if includesYou = true and memberCount > 50', () => {
    const component = shallowOption(
      { isSelected: true },
      buildTeam({
        includesYou: true,
        memberCount: 51,
      }),
    );
    expect(component).toMatchSnapshot();
  });
});
