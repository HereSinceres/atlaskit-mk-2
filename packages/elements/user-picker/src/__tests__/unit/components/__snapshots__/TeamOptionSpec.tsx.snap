// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Option should not render byline if member count is undefined 1`] = `
<Component
  avatar={
    <SizeableAvatar
      appearance="big"
      name="Team-1"
      src="https://avatars.atlassian.com/team-1.png"
    />
  }
  primaryText={
    Array [
      <styled.span
        color="#FFFFFF"
      >
        <HighlightText>
          Team-1
        </HighlightText>
      </styled.span>,
    ]
  }
/>
`;

exports[`Team Option should render correct byline if includesYou = false and memberCount <= 50 1`] = `
<Component
  avatar={
    <SizeableAvatar
      appearance="big"
      name="Team-1"
      src="https://avatars.atlassian.com/team-1.png"
    />
  }
  primaryText={
    Array [
      <styled.span
        color="#FFFFFF"
      >
        <HighlightText>
          Team-1
        </HighlightText>
      </styled.span>,
    ]
  }
  secondaryText={
    <styled.span
      color="#C1C7D0"
    >
      <FormattedMessage
        defaultMessage="{count} {count, plural, one {member} other {members}}"
        description="Byline to show the number of members in the team when the current user is not a member of the team"
        id="fabric.elements.user-picker.team.member.count"
        values={
          Object {
            "count": 45,
          }
        }
      />
    </styled.span>
  }
/>
`;

exports[`Team Option should render correct byline if includesYou = false and memberCount > 50 1`] = `
<Component
  avatar={
    <SizeableAvatar
      appearance="big"
      name="Team-1"
      src="https://avatars.atlassian.com/team-1.png"
    />
  }
  primaryText={
    Array [
      <styled.span
        color="#FFFFFF"
      >
        <HighlightText>
          Team-1
        </HighlightText>
      </styled.span>,
    ]
  }
  secondaryText={
    <styled.span
      color="#C1C7D0"
    >
      <FormattedMessage
        defaultMessage="50+ members"
        description="Byline to show the number of members in the team when the number exceeds 50"
        id="fabric.elements.user-picker.team.member.50plus"
        values={Object {}}
      />
    </styled.span>
  }
/>
`;

exports[`Team Option should render correct byline if includesYou = true and memberCount <= 50 1`] = `
<Component
  avatar={
    <SizeableAvatar
      appearance="big"
      name="Team-1"
      src="https://avatars.atlassian.com/team-1.png"
    />
  }
  primaryText={
    Array [
      <styled.span
        color="#FFFFFF"
      >
        <HighlightText>
          Team-1
        </HighlightText>
      </styled.span>,
    ]
  }
  secondaryText={
    <styled.span
      color="#C1C7D0"
    >
      <FormattedMessage
        defaultMessage="{count} {count, plural, one {member} other {members}}, including you"
        description="Byline to show the number of members in the team when the current user is also a member of the team"
        id="fabric.elements.user-picker.team.member.count.including.you"
        values={
          Object {
            "count": 45,
          }
        }
      />
    </styled.span>
  }
/>
`;

exports[`Team Option should render correct byline if includesYou = true and memberCount > 50 1`] = `
<Component
  avatar={
    <SizeableAvatar
      appearance="big"
      name="Team-1"
      src="https://avatars.atlassian.com/team-1.png"
    />
  }
  primaryText={
    Array [
      <styled.span
        color="#FFFFFF"
      >
        <HighlightText>
          Team-1
        </HighlightText>
      </styled.span>,
    ]
  }
  secondaryText={
    <styled.span
      color="#C1C7D0"
    >
      <FormattedMessage
        defaultMessage="50+ members, including you"
        description="Byline to show the number of members in the team when the number exceeds 50 and also includes the current user"
        id="fabric.elements.user-picker.team.member.50plus.including.you"
        values={Object {}}
      />
    </styled.span>
  }
/>
`;

exports[`Team Option should render correct byline if includesYou is undeifned and memberCount <= 50 1`] = `
<Component
  avatar={
    <SizeableAvatar
      appearance="big"
      name="Team-1"
      src="https://avatars.atlassian.com/team-1.png"
    />
  }
  primaryText={
    Array [
      <styled.span
        color="#FFFFFF"
      >
        <HighlightText>
          Team-1
        </HighlightText>
      </styled.span>,
    ]
  }
  secondaryText={
    <styled.span
      color="#C1C7D0"
    >
      <FormattedMessage
        defaultMessage="{count} {count, plural, one {member} other {members}}"
        description="Byline to show the number of members in the team when the current user is not a member of the team"
        id="fabric.elements.user-picker.team.member.count"
        values={
          Object {
            "count": 45,
          }
        }
      />
    </styled.span>
  }
/>
`;

exports[`Team Option should render correct byline if includesYou is undeifned and memberCount > 50 1`] = `
<Component
  avatar={
    <SizeableAvatar
      appearance="big"
      name="Team-1"
      src="https://avatars.atlassian.com/team-1.png"
    />
  }
  primaryText={
    Array [
      <styled.span
        color="#FFFFFF"
      >
        <HighlightText>
          Team-1
        </HighlightText>
      </styled.span>,
    ]
  }
  secondaryText={
    <styled.span
      color="#C1C7D0"
    >
      <FormattedMessage
        defaultMessage="50+ members"
        description="Byline to show the number of members in the team when the number exceeds 50"
        id="fabric.elements.user-picker.team.member.50plus"
        values={Object {}}
      />
    </styled.span>
  }
/>
`;
